.circleci/config.yml

```
@@ -212,6 +212,35 @@ jobs:
           tfp_ver: <<pipeline.parameters.max_tfp_ver>>
           pytest_filter: docs
 
+  build-notebooks:
+    # Note that /tmp/gpflow_build_docs is hard-coded a couple of times below. This path is hardcoded
+    # in doc/build_docs.py.
+    docker:
+      # At the time of writing we cannot generate documentation for type numpy with older versions,
+      # so we need to use `max` versions here.
+      - image: cimg/python:<<pipeline.parameters.max_py_ver>>
+        environment:
+          DOCS: True  # This is used in gpflow/ci_utils.py
+    parallelism: 10
+    steps:
+      - install_gpflow:
+          py_ver: <<pipeline.parameters.max_py_ver>>
+          tf_ver: <<pipeline.parameters.max_tf_ver>>
+          tfp_ver: <<pipeline.parameters.max_tfp_ver>>
+      - run:
+          name: Build notebooks
+          no_output_timeout: 60m
+          command: |
+            source .venv/bin/activate
+            mkdir -p /tmp/gpflow_build_docs
+            time python doc/build_docs.py --shard ${CIRCLE_NODE_INDEX}/${CIRCLE_NODE_TOTAL}
+            # Each shard must output unique files, so clean up output directory:
+            rm $(find /tmp/gpflow_build_docs -type f -and -not -name \*.ipynb)
```

I think you can do this (but don't quote me on it):

```suggestion
            find /tmp/gpflow_build_docs -type f -and -not -name \*.ipynb -delete
```

---

Good call. Seems to work. Thanks!